packages/shared/src/game_map.ts

```diff
@@ -18,8 +18,8 @@ class ConfigError extends Error {
 
 export function makeGameObject(
   variant: string,
-  config: any
-): AbstractGame<any, any> {
+  config: unknown
+): AbstractGame<unknown, unknown> {
```

Comment:
I had to look up unknown, but not sure if I understand. Would this create an error before compiling? (I assume so, but not sure)

const config: BadukConfig = { width: 9, height: 9, komi: 6 }
const game: AbstractGame<BadukConfig, BadukState> = makeGameObject("baduk", config)

Reply:
Exactly!  It basically says "don't make assumptions about what comes out of this function".  There are ways we could type this so that TypeScript knows that `makeGameObject("baduk", config)` always outputs a `Baduk`, but I think that kind of defeats the purpose of this function (we can just call `new Baduk(config)` instead)

You might have noticed I silenced a similar error in vue-client/src/store/baduk.ts using a type assertion:

```
function getStateFromResponse(response: GameResponse): BadukState {
  const game = makeGameObject(response.variant, response.config);
  response.moves.forEach((move) => {
    game.playMove(move);
  });
  return game.exportState() as BadukState;
}
```

I'm not terribly proud of that (as I said earlier - `as` is a way to trick the compiler), but I think the store stuff needs to be modified to handle multiple variants anyway.